habitat-lab/habitat/tasks/nav/nav.py

```diff
@@ -1344,6 +1348,13 @@ def _apply_velocity_action(
         # TODO: Make a better way to flag collisions
         self._sim._prev_sim_obs["collided"] = collided  # type: ignore
 
+        # Update the state of the agent
+        agent_observations = self._sim.get_observations_at(
+            position=final_position,
+            rotation=final_rotation,
+            keep_agent_at_new_pose=True,
+        )
```

Comment:
Is the idea here that if we don't call `self._sim.get_observations_at(..., keep_agent_at_new_pose=True)`, the agent doesn't actually move?
If so, we should deduplicate this with `get_agent_observation`.

Also one minor concern, does `self._sim.get_observations_at` incur additional rendering cost? Is that something we should be worried about? @ykarmesh

Reply:
Right, as my comment suggested, the agent does not move if you do not call self._sim.get_observations_at() to update the position. Yeah, we can reuse the function.